src/test/java/com/jcabi/xml/StrictXMLTest.java

```diff
@@ -168,27 +175,33 @@ public Void call() throws Exception {
 
     /**
      * Passes valid xml with network problems.
-     * @todo #32 This test does nothing useful for the moment, to be a real test
-     *  it needs to mock com.jcabi.xml.StrictXML.newValidator() and new mocked
-     *  validator should throw few SocketException's while executing
-     *  javax.xml.validation.Validator.validate(Source) in order to simulate the
-     *  situation with network problems. For more details, see
-     *  com.jcabi.xml.StrictXML.validate(XML) as well.
+     * @throws Exception If something goes wrong inside
      */
     @Test
-    public void passesValidXmlWithNetworkProblems() {
+    public void passesValidXmlWithNetworkProblems() throws Exception {
+        final Validator validator = Mockito.mock(Validator.class);
+        final AtomicInteger counter = new AtomicInteger(0);
+        Mockito.doAnswer(
+            new Answer<Void>() {
+                // @checkstyle IllegalThrowsCheck (5 lines)
+                @Override
+                public Void answer(final InvocationOnMock invocation)
+                    throws Throwable {
+                    final int attempt = counter.incrementAndGet();
+                    if (attempt == 1 || attempt == 2) {
+                        throw new SocketException(
+                            String.format("Attempt #%s failed", attempt)
+                        );
+                    } else {
+                        return null;
+                    }
+                }
+            }).when(validator).validate(Mockito.any(Source.class));
```

Comment:
I would put this closing bracket to the new line, see http://www.yegor256.com/2014/10/23/paired-brackets-notation.html